server-kotlin/src/city_sync/CitySynchronizationService.kt

```diff
@@ -0,0 +1,220 @@
+                        it[id] = event.id
+                        it[profileId] = newProfileId
+                        it[year] = event.year
+                        it[name] = event.name
+                        it[description] = event.description
+                    } get Events.id
+
+                    val eventDesc = event.eventDescription
+                    if (eventDesc != null) {
+                        EventDescriptions.insert {
+                            it[profileId] = newProfileId
+                            it[eventId] = eventsId
+                            it[year] = eventDesc.year
+                            it[category] = eventDesc.category
+                            it[eventName] = eventDesc.eventName
+                            it[organizationName] = eventDesc.organizationName
+                            it[description] = eventDesc.description
+                        }
+                    }
+                }
+
+                when (!cityExport.noticeboards.isNullOrEmpty()) {
+                    true -> Noticeboards.batchInsert(cityExport.noticeboards) {
+                        this[Noticeboards.profileId] = newProfileId
+                        this[Noticeboards.attachments] = it.attachments
+                        this[Noticeboards.category] = it.category
+                        this[Noticeboards.date] = it.date?.toDateTimeAtStartOfDay()
+                        this[Noticeboards.documentUrl] = it.documentUrl
+                        this[Noticeboards.edeskyUrl] = it.edeskyUrl
+                        this[Noticeboards.previewUrl] = it.previewUrl
+                    }
+                }
+
+                when (!cityExport.payments.isNullOrEmpty()) {
+                    true -> Payments.batchInsert(cityExport.payments) {
+                        this[Payments.profileId] = newProfileId
+                        this[Payments.amount] = it.amount
+                        this[Payments.counterpartyId] = it.counterpartyId
+                        this[Payments.counterpartyName] = it.counterpartyName
+                        this[Payments.date] = it.date?.toDateTimeAtStartOfDay()
+                        this[Payments.description] = it.description
+                        this[Payments.event] = it.event
+                        this[Payments.item] = it.item
+                        this[Payments.unit] = it.unit
+                        this[Payments.year] = it.year
+                        this[Payments.paragraph] = it.paragraph
+                    }
+                }
+                commit()
+            }
+        } catch (e: Exception) {
+            throw CitySyncException(message = e.message ?: "Unexpected error")
+        }
+    }
+
+    private suspend fun callInstance(syncTask: SyncTask): CityExport {
+        try {
+            val config =
+                HoconApplicationConfig(ConfigFactory.load()).property("app.city-sync.${syncTask.instance}").getString()
```

Comment:
any chance to pass this as a dependency?  The config is already resolved in Application.kt

Reply:
I will take a look at it tomorrow or by the weekend. When I wrote this, there were no other implementation of configs :)

Reply:
I've looked at the way config is used in Application.kt, but personally I think it would be better to create a config class, that would be passed as a param to a router or service class instead od a bunch of strings. What do you think?

Reply:
👍 agree

Reply:
@michalsvec I made a CommonConfig class, please have a look at it. If you do I'll merge it into this PR https://github.com/HonzaV/cityvizor/commit/67d23bb717b563fde474af006ebb1f48c32c26d3

Reply:
I like that

Reply:
Ok, merged